Replace custom notification sound with system sound

gasi opened the pull request:
- [x] Remove existing custom sound (aka Ka-Klunk).
- [x] Use system sound on macOS and Windows.
- [x] Defer sound settings to OS on Linux.
- [x] Persist flashing of taskbar frame on Windows.

Fixes #1554.
Fixes #1679.

gasi commented:
@scottnonnenberg: Would you mind taking a look at this PR whenever you have time after the release? Fixing the task bar flashing on Windows might help the following two users: https://github.com/signalapp/Signal-Desktop/issues/1679#issuecomment-368967619 Thanks 😄

scottnonnenberg commented on js/notifications.js:
```diff
@@ -27,15 +27,23 @@
         update: function() {
             const isFocused = window.isFocused();
+            const shouldPlayNotificationSound = Settings.isAudioNotificationSupported() &&
+                (storage.get('audio-notification') || false);
```
Might be good to introduce an explaining variable for this line, given the

scottnonnenberg commented on js/notifications.js:
```diff
@@ -27,15 +27,23 @@
         update: function() {
             const isFocused = window.isFocused();
+            const shouldPlayNotificationSound = Settings.isAudioNotificationSupported() &&
+                (storage.get('audio-notification') || false);
```
Might be good to introduce an explaining variable for this line, given the `|| false`

gasi replied:
Good idea! Fixed it.

scottnonnenberg commented on js/notifications.js:
```diff
@@ -61,9 +64,13 @@
             var message;
             var iconUrl;
 
+            // NOTE: i18n has more complex rules for pluralization than just
```
Probably not worth putting this here.

scottnonnenberg commented on js/notifications.js:
```diff
@@ -92,7 +99,7 @@
                     title: title,
                     message: message,
-                    sound: false
+                    sound: shouldPlayNotificationSound,
```
This is interesting. We only use `nodeNotifier` on Windows 7. So I presume we haven't tested this. What's the theory, at least?

gasi replied:
Good call, I reverted it since I couldn’t test it! The docs say this is macOS only and we never polyfill notifications there. I’ll get back to this whenever I get a chance to set up a full Windows 7 VM. Or maybe someone from the community can play around with this and let us know what the Windows 7 behavior is 😄

gasi commented:
Rebased and addressed requested changes: https://github.com/signalapp/Signal-Desktop/pull/2071/files/5573ee58c6474325d4abdc1e94a43998030abc8b..3b577c3dc0175f3e80f56r7d3089c645abc730789

gasi commented on js/notifications.js:
```diff
@@ -27,15 +27,27 @@
+                'numNotifications:', numNotifications,
+                'isAudioNotificationEnabled:', isAudioNotificationEnabled,
+                'isAudioNotificationSupported:', isAudioNotificationSupported,
+                'shouldPlayNotificationSound:', shouldPlayNotificationSound
```
Good call! Too bad that we have to be careful with log size, but I am happy to take out `isAudioNotificationEnabled` and `isAudioNotificationSupported`. That will leave `shouldPlayNotificationSound` which combines the two. Sound good to you?

gasi commented on js/notifications.js:
```diff
@@ -122,14 +129,14 @@
-            var update = !enabled;
-            enabled = true;
-            if (update) {
+            var shouldUpdate = !isEnabled;
```
Sounds good, I’ll do `needUpdate`.

gasi commented on js/notifications.js:
```diff
@@ -122,14 +129,14 @@
         },
         disable: function() {
-            enabled = false;
+            isEnabled = false;
```
Totally agreed! I wasn’t sure why it was designed like this so I left it but I can make it an instance variable and re-test quickly.

gasi replied:
FWIW, we treat it as a singleton: https://github.com/signalapp/Signal-Desktop/blob/40ee88ead31a3d7c6ee32767f9e7e5c1ba8dce71/js/background.js#L387

gasi commented:
Thanks for your feedback. Pushed my updates to it (and didn’t rebase 😉) Did you get the GitHub change indicator?

gasi commented:
If not, here’s the diff: https://github.com/signalapp/Signal-Desktop/pull/2071/files/05b695b3028aa60488f6f92b7ef1e3207979fa6d..59b6807d4538141e3174c5d7644f1337d93eaf26

How to easily get that link: http://recordit.co/GoprWIIX4i

gasi commented:
This branch got out of sync due to merge before pushing rebased changes. Fixing it in https://github.com/signalapp/Signal-Desktop/pull/2108 😄
